### PR TITLE
[Owl] Fixes multiple navigation calls with simultaneous taps

### PR DESCRIPTION
**Pull request by JoseAlcerreca**

### CODE REVIEW

File: Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt [hunk 1]
```diff
@@ -83,7 +84,7 @@ fun NavGraph(
             startDestination = CourseTabs.FEATURED.route
         ) {
             courses(
-                onCourseSelected = actions.selectCourse,
+                onCourseSelected = { newCourseId: Long -> actions.openCourse(newCourseId) },
```
Comment (JoseAlcerreca) [inline Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt:87]: Revert

File: Owl/app/src/main/java/com/example/owl/ui/course/CourseDetails.kt [hunk 2]
```diff
@@ -323,7 +324,10 @@ private fun RelatedCourses(
                     items(relatedCourses) { related ->
                         CourseListItem(
                             course = related,
-                            onClick = { selectCourse(related.id) },
+                            onClick = {
+                                Log.d("jalc", "jalc")
```
Comment (manuelvicnt) [inline Owl/app/src/main/java/com/example/owl/ui/course/CourseDetails.kt:328]: Nit: remove log

File: Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt [hunk 3]
```diff
@@ -112,10 +117,31 @@ class MainActions(navController: NavHostController) {
...
+    // Used from COURSE_DETAIL_ROUTE
+    val upPress: (rom: NavBackStackEntry) -> Unit = { from ->
+        // In order to discard duplicated navigation events, we check the Lifecycle
+        if (from.lifecycleIsResumed()) {
+            navController.navigateUp()
+        }
     }
 }
+
+private fun NavBackStackEntry.lifecycleIsResumed() =
```
Comment (manuelvicnt) [inline Owl/app/src/main/java/com/example/owl/ui/NavGraph.kt:146]: Nit: Can you add a note saying why this is needed when navigating between screens? Is this always needed?

File: Owl/app/src/main/java/com/example/owl/ui/course/CourseDetails.kt [hunk 4]
```diff
@@ -16,6 +16,7 @@
 
 package com.example.owl.ui.course
 
+import android.util.Log
```
Comment (manuelvicnt) [inline Owl/app/src/main/java/com/example/owl/ui/course/CourseDetails.kt:19]: Nit: remove

**Review by manuelvicnt — APPROVED** [review]

**Pull request merged**